controllers/feeds/getOneFeedChat.js

```diff
@@ -83,6 +84,34 @@ const getOneFeedChatService = async (req, res) => {
     }
 
     const actor = data?.results[0]?.actor;
+
+    let karmaActors = [];
+    if (!karmaActors.includes(actor.id)) {
+      karmaActors.push(actor.id);
+    }
+    for (const comment of comments) {
+      if (!karmaActors.includes(comment?.reaction?.user_id)) {
+        karmaActors.push(comment?.reaction?.user_id);
+      }
+    }
+
+    const karmaScores = await UsersFunction.getUsersKarmaScore(User, karmaActors);
+    comments = comments.map((comment) => {
+      const user = karmaScores.find((user) => user.user_id === comment.reaction.user_id);
+      comment.reaction.user.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+      comment.reaction.latest_children.comment = comment.reaction?.latest_children?.comment?.map(
+        (child) => {
+          const user = karmaScores.find((user) => user.user_id === child.user_id);
+          child.user.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+          return child;
+        }
+      );
+      return comment;
+    });
+
+    const user = karmaScores.find((user) => user.user_id === actor.id);
+    actor.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for fetching and assigning karma scores seems correct. However, there's a potential performance issue here. For each comment, you're searching through the `karmaScores` array to find the corresponding user. This results in a time complexity of O(n^2) which could be problematic if there are many comments and users. 

A more efficient approach would be to convert `karmaScores` into an object where the keys are the user IDs and the values are the karma scores. This way, you can look up the karma score for a user in constant time.

Here's how you can implement this:

```diff
- const karmaScores = await UsersFunction.getUsersKarmaScore(User, karmaActors);
+ const karmaScoresArray = await UsersFunction.getUsersKarmaScore(User, karmaActors);
+ const karmaScores = {};
+ for (const user of karmaScoresArray) {
+   karmaScores[user.user_id] = user.karma_score;
+ }

  comments = comments.map((comment) => {
-   const user = karmaScores.find((user) => user.user_id === comment.reaction.user_id);
+   const userKarmaScore = karmaScores[comment.reaction.user_id] || 0;
-   comment.reaction.user.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+   comment.reaction.user.karmaScores = roundingKarmaScore(userKarmaScore);

    comment.reaction.latest_children.comment = comment.reaction?.latest_children?.comment?.map(
      (child) => {
-       const user = karmaScores.find((user) => user.user_id === child.user_id);
+       const userKarmaScore = karmaScores[child.user_id] || 0;
-       child.user.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+       child.user.karmaScores = roundingKarmaScore(userKarmaScore);
        return child;
      }
    );
    return comment;
  });

- const user = karmaScores.find((user) => user.user_id === actor.id);
+ const userKarmaScore = karmaScores[actor.id] || 0;
- actor.karmaScores = roundingKarmaScore(user?.karma_score || 0);
+ actor.karmaScores = roundingKarmaScore(userKarmaScore);
```
This change will significantly improve the performance of your code when dealing with large numbers of comments and users.


<!-- This is an auto-generated comment by OSS CodeRabbit -->